astroid/brain/brain_typing.py

```diff
@@ -344,6 +345,29 @@ def infer_tuple_alias(
     return iter([class_def])
 
 
+def _looks_like_typing_cast(node: Call) -> bool:
+    return (
+        isinstance(node, Call)
+        and isinstance(node.func, Name)
+        and node.func.name == "cast"
+    )
+
+
+def infer_typing_cast(
+    node: Call, ctx: context.InferenceContext = None
+) -> typing.Iterator[Instance]:
+    """Infer call to cast() returning an instance of the cast-to type"""
+    try:
+        func = next(node.func.infer(context=ctx))
+    except InferenceError as exc:
+        raise UseInferenceDefault from exc
+    if func.qname() != "typing.cast":
```

Comment:
```suggestion
    if func.qname() != "typing.cast" and len(node.args) == 2:
```
Just to prevent errors if the cast statement syntax isn't correct.

Reply:
Good point, but I think this should be `func.qname() != "typing.cast" or len(node.args) != 2`, unless I've misunderstood?

Reply:
My mistake. Yours is correct.